### PR TITLE
[TASK] Modernize Unit test examples

### PR DESCRIPTION
**Pull request by linawolf**
Releases: main, 12.4

### CODE REVIEW

File: Documentation/Testing/WritingUnit.rst [hunk 1]
```diff
@@ -99,7 +99,8 @@ by creating a functional test from it.
 Unit test conventions
 =====================
 
-TYPO3 unit testing means using the `phpunit <https://phpunit.de/>`_ testing framework. TYPO3 comes with
+TYPO3 unit testing means using the `phpunit <https://phpunit.de/>`_ testing
+framework. The TYPO3 testing framework comes with
 as basic `UnitTests.xml <https://github.com/TYPO3/testing-framework/blob/main/Resources/Core/Build/UnitTests.xml>`_
```
Comment (garvinhicking) [inline Documentation/Testing/WritingUnit.rst:104]:
```suggestion
a basic `UnitTests.xml <https://github.com/TYPO3/testing-framework/blob/main/Resources/Core/Build/UnitTests.xml>`_
```

**Review by garvinhicking — COMMENTED** [review]
Can't comment much on the rest ATM sadly!

File: Documentation/Testing/_UnitTests/_SomeTest.php [hunk 2]
```diff
@@ -10,7 +10,14 @@ class SomeTest extends UnitTestCase
...
-        $iconFactoryProphecy = $this->prophesize(IconFactory::class);
-        GeneralUtility::addInstance(IconFactory::class, $iconFactoryProphecy->reveal());
+        $iconFactory =
+            $this->createMock(IconFactory::class);
+        GeneralUtility::addInstance(IconFactory::class, $iconFactory);
+    }
+
+    protected function tearDown(): void
+    {
+        GeneralUtility::purgeInstances();
```
Comment (froemken) [inline Documentation/Testing/_UnitTests/_SomeTest.php:20]: Very good

File: Documentation/Testing/_UnitTests/_TcaFlexPrepareTest.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+use PHPUnit\Framework\Attributes\Test;
+use TYPO3\CMS\Core\Cache\CacheManager;
+use TYPO3\CMS\Core\Cache\Frontend\FrontendInterface;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\TestingFramework\Core\Unit\UnitTestCase;
+
+final class TcaFlexPrepareTest extends UnitTestCase
+{
+    protected bool $resetSingletonInstances = true;
```
Comment (froemken) [inline Documentation/Testing/_UnitTests/_TcaFlexPrepareTest.php:13]: nice

**Review by froemken — APPROVED** [review]

**Pull request merged**